src/NerdBank.GitVersioning/VersionOracle.cs

```diff
@@ -296,17 +297,29 @@ public IEnumerable<string> BuildMetadataWithCommitId
         {
             var variables = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-            PropertyInfo[]? properties = this.GetType().GetTypeInfo().GetProperties(BindingFlags.Public | BindingFlags.DeclaredOnly | BindingFlags.Instance);
-            foreach (PropertyInfo? property in properties)
+            PropertyInfo[] properties = this.GetType().GetTypeInfo().GetProperties(BindingFlags.Public | BindingFlags.DeclaredOnly | BindingFlags.Instance);
+            foreach (PropertyInfo property in properties)
             {
-                if (property.GetCustomAttribute<IgnoreAttribute>() is null)
+                if (property.GetCustomAttribute<IgnoreAttribute>() is not null)
+                {
+                    continue;
+                }
+
+                object? propertyValue = property.GetValue(this);
+                if (propertyValue is null)
                 {
-                    object? value = property.GetValue(this);
-                    if (value is object)
-                    {
-                        variables.Add($"NBGV_{property.Name}", value.ToString() ?? string.Empty);
-                    }
+                    continue;
                 }
+
+                const string isoDateTimeFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFFK";
+
+                string value = propertyValue switch
+                {
+                    DateTimeOffset dateTimeOffset => dateTimeOffset.ToString(isoDateTimeFormat, CultureInfo.InvariantCulture),
```

Comment:
No need to hard-code the ISO format. Just use `o`, which [means the same thing](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#the-round-trip-o-o-format-specifier).

```suggestion
                string value = propertyValue switch
                {
                    DateTimeOffset dateTimeOffset => dateTimeOffset.ToString("o", CultureInfo.InvariantCulture),
```